src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -430,6 +431,43 @@ void optimizeGroupByFunctionKeys(ASTSelectQuery * select_query, bool optimize_gr
     grp_by->children = modified;
 }
 
+/// Eliminates min/max/any-aggregators of functions of GROUP BY keys
+void optimizeAggregateFunctionsOfGroupByKeys(ASTSelectQuery * select_query, bool optimize_aggregators_of_group_by_keys)
+{
+    if (!optimize_aggregators_of_group_by_keys)
+        return;
+
+    if (!select_query->groupBy())
+        return;
+
+    auto grp_by = select_query->groupBy();
+    auto & group_keys = grp_by->children;
+
+    std::unordered_set<String> group_by_keys; ///set of keys' short names
+
+    ///filling set with short names of keys
+    for (auto & group_key : group_keys)
```

Comment:
There's almost the same code in `optimizeGroupByFunctionKeys()`. We need a function that extracts such ckeys instead of code duplication